plugins/eos-blacklist/gs-plugin-eos-blacklist.c

```diff
@@ -207,19 +215,71 @@ app_get_flatpak_ref_name (GsApp *app)
 	return gs_app_get_metadata_item (app, "flatpak::RefName");
 }
 
+static gchar **
+get_flatpak_default_locales (GsPlugin *plugin)
+{
+	g_auto(GStrv) flatpak_default_locales = NULL;
+	g_autoptr(GError) local_error = NULL;
+
+	GsPluginData *priv = gs_plugin_get_data (plugin);
+	if (priv->installation == NULL)
+		return NULL;
+
+	flatpak_default_locales = flatpak_installation_get_default_locales (priv->installation, &local_error);
+	if (local_error != NULL) {
+		g_warning ("No user locales could be retrieved! %s", local_error->message);
+		return NULL;
+	}
+
+	return g_steal_pointer (&flatpak_default_locales);
+}
+
+static gboolean
+assert_valid_locale_match (const char *app_locale,
+                           char **locale_options)
+{
+	int idx;
+	g_auto(GStrv) app_locale_parts = g_strsplit (app_locale, "_", -1);
+
+	for (idx = 0; locale_options[idx] != 0; idx++) {
+		g_auto(GStrv) locale_parts = g_strsplit (locale_options[idx], "_", -1);
+		if (g_strcmp0 (app_locale_parts[0], locale_parts[0]) == 0) {
+			/* If the main is a match, try to match modifiers: eg. region, alphabet */
```

Comment:
Please use the same terminology as `man setlocale`, i.e. ‘territory’ and ‘codeset’ — locales are confusing enough already without introducing alternative terminology!